src/log/log.c

```diff
@@ -1970,6 +1970,10 @@ __log_write_internal(WT_SESSION_IMPL *session, WT_ITEM *record, WT_LSN *lsnp,
 	    myslot.slot != NULL)
 		ret = myslot.slot->slot_error;
 
+	if (LF_ISSET(WT_LOG_FLUSH))
+		WT_ASSERT(session, __wt_log_cmp(&log->write_lsn, &lsn) >= 0);
+	if (LF_ISSET(WT_LOG_FSYNC))
+		WT_ASSERT(session, __wt_log_cmp(&log->sync_lsn, &lsn) >= 0);
```

Comment:
I like that idea and have pushed that fix.  Once it says it can be merged I'll do so.